### ISSUE TITLE
Ignoring the result of String::replace should raise a warning

### ISSUE THREAD

**Issue by gnzlbg** [rust-lang/rust#48423]
[I was doing this (playground):](https://play.rust-lang.org/?gist=d82f860d28161a42f9634c850a273483&version=stable)

```rust
fn replace_slashes(s: &mut String) {
    s.replace(r#"\\"#, r#"\"#);
}

fn main() {
    let mut windows_path =
        r#"C:\\projects\\cargo-asm\\cargo-asm-test\\lib_crate\\src\\bar.rs"#.to_string();
    let windows_path_norm =
        r#"C:\projects\cargo-asm\cargo-asm-test\lib_crate\src\bar.rs"#.to_string();
    replace_slashes(&mut windows_path);
    assert_eq!(windows_path_norm, windows_path);
}
```

Spot the logic bug. Why would anybody call `String::replace` and ignore the result? This code should at least raise a warning, and the result of `String::replace` should probably be `#[must_use]` . Thinking that`String::replace` might work _in place_ is plausible.

**Comment by vitalyd** [comment]
Note that it’s `str::replace`, not `String`.  It also takes `&self`, which indicates it doesn’t mutate itself.

I do agree a `#[must_use]` would be helpful.  A clippy lint might do as well.

**Comment by varkor** [comment]
[`str::replace` now has a `#[must_use]` attribute](https://github.com/rust-lang/rust/blob/3d28ee3e34d100534062d0bc690779ed9b6927fe/src/liballoc/str.rs#L264-L265), so this issue can be closed.
